File: bigseqkit/head.go
package bigseqkit

import (
	"ignis/driver/api"
	"strconv"
)

type SeqKitHeadOptions struct {
	inner HeadOptions
}

type HeadOptions struct {
	Config KitConfig
	N      *int64
}

func (this *HeadOptions) setDefaults() *HeadOptions {
	this.Config.setDefaults()
	setDefault(&this.N, 10)

	return this
}

func (this *SeqKitHeadOptions) Config(v *SeqKitConfig) *SeqKitHeadOptions {
	this.inner.Config = v.inner
	return this
}

func (this *SeqKitHeadOptions) N(v int64) *SeqKitHeadOptions {
	this.inner.N = &v
	return this
}

func Head(input *api.IDataFrame[string], o *SeqKitHeadOptions) (*api.IDataFrame[string], error) {
	if o == nil {
		o = &SeqKitHeadOptions{}
	}
	opts := o.inner
	opts.setDefaults()

	oRange := (&SeqKitRangeOptions{}).Range("1:" + strconv.FormatInt(*opts.N, 10))
	oRange.inner.Config = opts.Config

	return Range(input, oRange)
}

File: bigseqkit-py/bigseqkit/head.py
from bigseqkit.helper import _setDefault, _libSource, _config, _optionsToString, _parseKargs, SeqKitConfig, IDataFrame
from bigseqkit.range import SeqKitRangeOptions, range

class SeqKitHeadOptions:

    def __init__(self):
        self.__inner = HeadOptions()

    def config(self, v: SeqKitConfig):
        self.__inner.Config = _config(v)

    def n(self, v: int):
        self.__inner.N = v

    def _run(self, input: IDataFrame, **kwargs):
        opts = self.__inner
        _parseKargs(opts, kwargs)
        opts.setDefaults()

        oRange = SeqKitRangeOptions().range("1:"+str(opts.N))
        oRange._SeqKitRangeOptions__inner = opts.Config

        return range(input, oRange)

class HeadOptions:

    def __init__(self):
        self.Config = None  # KitConfig
        self.N = None  # int

    def setDefaults(self):
        _setDefault(self, "Config", _config(SeqKitConfig())).setDefaults()
        _setDefault(self, "N", 10)


def head(input: IDataFrame, o: SeqKitHeadOptions = None, **kwargs):
    if o is None:
        o = SeqKitHeadOptions()
    return o._run(input, **kwargs)
